superset-frontend/src/dashboard/components/nativeFilters/FilterConfigModal/utils.ts

```diff
@@ -175,6 +177,30 @@ export const setFilterFieldValues = (
   });
 };
 
+const secondaryFields = ['defaultValueQueriesData', 'defaultValueFormData'];
+
+export const filterOutSecondaryFields = (formInputs: NativeFiltersFormItem) =>
+  Object.entries(formInputs).reduce((resultInFormInputs, [key, value]) => {
+    if (!secondaryFields.includes(key)) {
+      return {
+        ...resultInFormInputs,
+        [key]: value,
+      };
+    }
+    return resultInFormInputs;
+  }, {} as NativeFiltersFormItem);
```

Comment:
To avoid having to do this, would it make sense to introduce a new prop `controlValues` in which we place all control values that are specific to the filter at hand? In the case of the Select filter, it would then contain `enableEmptyFilter`, `inverseSelection` and `multiSelect`. It would also minimize the risk of name collision.

Reply:
done